### ISSUE TITLE
unicode filename problem on both gnome and kde

### ISSUE THREAD

**Issue by gokhanmoral** [jstaf/onedriver#390]
files or folders with unicode characters cannot be fetched or used.

reproduction method:
creating a file over onedrive web with the name "THIS_IS_TEST.txt" with content "test1" can be displayed with gnome text editor or kwrite.
creating a file over onedrive web with the name "THİS_İS_TEST.txt" with content "test2" cannot be opened.

creating a folder with letter "İ" in onedriver folder creates a folder on onedrive and it can be used over web, but the same folder cannot be used on neither gnome files nor dolphin.

**Comment by gokhanmoral** [comment]
additonal info:

on terminal, somehow autocompletion works on fetching the file. but when I type in the filename manually, file cannot be fetched.

```
gm@fedora:~/onedrive$ cat THİS_İS_TEST3.txt 
test3
gm@fedora:~/onedrive$ cat THİS_İS_TEST3.txt
cat: THİS_İS_TEST3.txt: No such file or directory
gm@fedora:~/onedrive$ 
```
in the first one, the file is selected with auto completion (writing 'TH' followed by a tab press)
in the second one, I typed the filename using Turkish keyboard.

**Comment by foopsss** [comment]
Is the "İ" character a turkish one? Either way, I think this **might** be related to the naming conventions issue mentioned in #387.

**Comment by gokhanmoral** [comment]
> Is the "İ" character a turkish one? Either way, I think this **might** be related to the naming conventions issue mentioned in #387.

yes, it is a turkish character which is not available in ISO-8859-1 but it is in ISO-8859-9. and no, it is not a disallowed character in none of the windows filesystems. it is usable in ntfs as well as onedrive. I suspect that the representation differences in utf-8, utf-16 and single char using 8850p9.

**Comment by foopsss** [comment]
Hmmph, I see. Hopefully @jstaf or @abraunegg have something else to say on this regard.

**Comment by abraunegg** [comment]
@foopsss , @gokhanmoral 
I cannot comment on how 'onedriver' operates.

The 'onedrive' application I maintain and develop checks compliance against a number of specific items (as per current development tree 'alpha-5'):
* Check path against Microsoft OneDrive restriction and limitations about Windows naming for files and folders
* Check path for bad whitespace items
* Check path for HTML ASCII Codes
* Check path for ASCII Control Codes
* Validate that the path is a valid UTF-16 (ISO/IEC 10646) encoded path & does not contain invalid UFT sequences
